src/main/java/jenkins/plugins/hipchat/workflow/HipChatStep.java

```diff
@@ -0,0 +1,129 @@
+    public HipChatStep(@Nonnull String message) {
+        this.message = message;
+    }
+
+    @Extension
+    public static class DescriptorImpl extends AbstractStepDescriptorImpl {
+
+        public DescriptorImpl() {
+            super(HipChatStepExecution.class);
+        }
+
+        @Override
+        public String getFunctionName() {
+            return "hipchat";
+        }
+
+        @Override
+        public String getDisplayName() {
+            return "Publish HipChat Message";
+        }
+
+        public ListBoxModel doFillColorItems(){
+            ListBoxModel listBoxModel = new ListBoxModel();
+            for(Color color : Color.values()) {
+                listBoxModel.add(new ListBoxModel.Option(color.toString(), color.toString(), false));
+            }
+
+           return listBoxModel;
+        }
+
+    }
+
+    public static class HipChatStepExecution extends AbstractSynchronousNonBlockingStepExecution<Void> {
+
+        private static final long serialVersionUID = 1L;
+
+        @Inject
+        private transient HipChatStep step;
+        @StepContextParameter
+        transient TaskListener listener;
+
+
+        @Override
+        protected Void run() throws Exception {
+
+            if (StringUtils.isBlank(step.message)) {
+                throw new AbortException(Messages.MessageRequiredError());
+            }
+
+            //default to global config values if not set in step, but allow step to override all global settings
+            HipChatNotifier.DescriptorImpl hipChatDesc = (HipChatNotifier.DescriptorImpl) Jenkins.getInstance().getDescriptor(HipChatNotifier.class);
+            String token = step.token != null ? step.token : hipChatDesc.getToken();
+            String room = step.room != null ? step.room : hipChatDesc.getRoom();
+            String server = step.server != null ? step.server : hipChatDesc.getServer();
+            String sendAs = step.sendAs != null ? step.sendAs : hipChatDesc.getSendAs();
+            //default to gray if not set in step
+            String color = step.color != null ? step.color : Color.GRAY.toString();
+            boolean v2enabled = step.v2enabled != null ? step.v2enabled : hipChatDesc.isV2Enabled();
+
+            //only way to use this static method from HipChatNotifier and keep HipChatStep in the workflow package is to make it public
+            HipChatService hipChatService = HipChatNotifier.getHipChatService(server, token, v2enabled, room, sendAs);
+
+            //placing in console log to simplify testing of retrieving values from global config or from step field
+            listener.getLogger().println(Messages.WorkflowStepConfig(step.server == null, step.token == null, step.room == null, step.color == null));
+
+            logger.finer("HipChat publish settings: api v2 - " + v2enabled + " server - " +
+                    server + " token - " + token + " room - " + room);
+
+            hipChatService.publish(step.message, color, step.notify);
```

Comment:
should you log something to tasklistener if publish threw a NotificationException?